packages/rum/src/domain/record/trackers/trackScroll.ts

```diff
@@ -18,7 +18,8 @@ export type ScrollCallback = (incrementalSnapshotRecord: BrowserIncrementalSnaps
 export function trackScroll(
   configuration: RumConfiguration,
   scrollCb: ScrollCallback,
-  elementsScrollPositions: ElementsScrollPositions
+  elementsScrollPositions: ElementsScrollPositions,
+  target: Document | ShadowRoot = document
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏instead of having document as default, what about having a unit test to ensure that it works with both types of target?

Reply:
Agreed on DM that since the logic is the same for tracking scrolls on document, elements and shadowRoot, a unit test is not needed.
The change is covered by an e2e tests.

Reply:
> Agreed on DM that since the logic is the same for tracking scrolls on document, elements and shadowRoot, a unit test is not needed.

I just did not had time to answer back 🙂
My point was that since we are introducing a new param, it could be valuable to test that this function behave as expected for different values of this param.

By looking more at it, it could be nice to ensure that with trackScroll on different targets, only the expected tracker call its callback.

 > The change is covered by an e2e tests.

it should not prevent us to add additional unit tests.
To be noted: more browsers are covered with unit tests.

Reply:
Added an integration test ✔️